dask/order.py

```diff
@@ -349,6 +349,15 @@ def dependencies_key(x):
         while item in result:
             item = init_stack_pop()
 
+    # TODO: Hack to priorities "shuffle-split"
+    shuffle_split_keys = []
+    for k in result.keys():
+        if len(k) > 0 and "shuffle-split" in k[0]:
```

Comment:
What about using a general label that any API or user could add to the key of a task?  For example, something like `"dsk-prioritize"`:

```python
# Prioritize tasks with "dsk-prioritize" annotation
for k in list(result.keys()):
    if k and "dsk-prioritize" in k[0]:
        result[k] = 0
```

I know you mentioned off-line that the best general solution would be to annotate the priority within the task.  Is this what you had in mind - or something more sophisticated?  I guess the specific priority could also be included in the annotation if the user/api wants really fine-grained control.

Reply:
Yes, something like this but _overloading_ key names is properly too much of a hack to be accepted :)